app/src/main/java/com/andrew/tasky/agenda/presentation/screens/event_detail/EventDetailViewModel.kt

```diff
@@ -255,33 +211,35 @@ class EventDetailViewModel @Inject constructor(
                         )
                     )
                 }
-                val event = AgendaItem.Event(
-                    id = savedStateHandle.get<AgendaItem.Event>("event")?.id
-                        ?: UUID.randomUUID().toString(),
-                    isDone = isDone.value,
-                    title = title.value,
-                    description = description.value,
-                    startDateAndTime = LocalDateTime.of(
-                        selectedStartDate.value,
-                        selectedStartTime.value
-                    ),
-                    endDateAndTime = LocalDateTime.of(
-                        selectedEndDate.value,
-                        selectedEndTime.value
-                    ),
-                    reminderTime = selectedReminderTime.value,
-                    photos = eventPhotos,
-                    attendees = attendees.value,
-                    isCreator = savedStateHandle.get<AgendaItem.Event>("event")?.isCreator ?: true,
-                    host = savedStateHandle.get<AgendaItem.Event>("event")?.host,
-                    deletedPhotoKeys = emptyList(), // TODO setup deletedPhotosKeys
-                    isGoing = savedStateHandle.get<AgendaItem.Event>("event")?.isGoing ?: true
-                )
-                repository.upsertEvent(event)
             }
         }
     }
 
+    private fun getEvent(): AgendaItem.Event {
+        return AgendaItem.Event(
+            id = savedStateHandle.get<AgendaItem.Event>("event")?.id
+                ?: UUID.randomUUID().toString(),
+            isDone = isDone.value,
+            title = title.value,
+            description = description.value,
+            startDateAndTime = LocalDateTime.of(
+                selectedStartDate.value,
+                selectedStartTime.value
+            ),
+            endDateAndTime = LocalDateTime.of(
+                selectedEndDate.value,
+                selectedEndTime.value
+            ),
+            reminderTime = selectedReminderTime.value,
+            photos = photos.value,
+            attendees = attendees.value,
+            isCreator = savedStateHandle.get<AgendaItem.Event>("event")?.isCreator ?: true,
+            host = savedStateHandle.get<AgendaItem.Event>("event")?.host,
+            deletedPhotoKeys = emptyList(), // TODO setup deletedPhotosKeys
+            isGoing = savedStateHandle.get<AgendaItem.Event>("event")?.isGoing ?: true
```

Comment:
I'd avoid passing the event as a `Parcelable` to the event detail screen. With that, you're violating the single source of truth principle. In your app, the single source is your local DB. If you load from multiple sources (here from savedStateHandle for example), you're likely going to run into bugs. Imagine, your AgendaScreen didn't fully update its items yet, but there was a change in your DB for the event the user clicks on. Then if you click on the event on the AgendaScreen (which is outdated), you'll get an outdated detail view and if the user then saves it, it'll override the new event with the outdated one. 
Therefore, for detail screens in most cases I'd just pass the ID of the event and load it from the DB again.